docs/source/architecture/events_and_transactions_receipts.rst

```diff
@@ -0,0 +1,343 @@
+the subscriber's list of subscriptions.
+
+To reduce the number of messages sent to subscribers, multiple Event messages
+will be wrapped in an EventList message when possible:
+
+.. code-block:: python
+
+  EventList {
+    repeated Event events = 1;
+  }
+
+ClientEventSubscribeRequest are sent by external clients to the validator in
+order to subscribe to events. ClientEventSubscribeResponse are sent by the
+validator to the client in response to notify the client whether their
+subscription was successful. When an external client subscribes to events, they
+may optionally send a list of block ids along with their subscriptions. If any
+of the blocks sent are in the current chain, the EventBroadcaster will bring the
+client up to date by sending it events for all blocks since the most recent
+block sent with the subscribe request.
+
+.. code-block:: protobuf
+
+  message ClientEventsSubscribeRequest {
+      repeated EventSubscription subscriptions = 1;
+      // The block id (or ids, if trying to walk back a fork) the subscriber last
+      // received events on. It can be set to empty if it has not yet received the
+      // genesis block.
+      repeated string last_known_block_ids = 2;
+  }
+
+  message ClientEventsSubscribeResponse {
+      enum Status {
+           OK = 0;
+           INVALID_FILTER = 1;
+           UNKNOWN_BLOCK = 2;
+      }
+      Status status = 1;
+      // Additional information about the response status
+      string response_message = 2;
+  }
+
+Event Extractors
+----------------
+
+Two event extractors will be created to extract events from blocks and
+transaction receipts: BlockEventExtractor and ReceiptEventExtractor. The
+BlockEventExtractor will extract events of type "block_commit" and
+"batch_commit". The ReceiptEventExtractor will extract events of type
+"state_delta" and events defined by transaction families.
+
+Example events generated by BlockEventExtractor:
+
+.. code-block:: protobuf
+
+  // Example block_commit event
+  Event {
+    event_type = "block_commit",
+    meta_data = [
+      Meta { key = "block_id", value = "abc...123" },
+      Meta { key = "block_num", value = "523" },
+      Meta { key = "state_root_hash", value = "def...456" },
+      Meta { key = "previous_block_id", value = "acf...146" },
+    ],
+  }
+
+  // Example block_commit event
+  Event {
+    event_type = "batch_commit",
+    meta_data = [Meta { key = "batch_id", value = "abc...123" }],
+  }
+
+Example events generated by ReceiptEventExtractor:
+
+.. code-block:: protobuf
+
+
+  // Example transaction family specific event
+  Event {
+    event_type = "xo_create",
+    meta_data = [Meta { key = "game_name", value = "game0" }],
+  }
+
+  // Example block_commit event
+  Event {
+    event_type = "state_delta",
+    meta_data = [Meta { key = "address", value = "abc...def" }],
+    event_data = <bytes>
+  }
+
+
+Transaction Receipts
+====================
+
+Transaction receipts support the following use cases:
```

Comment:
I'd recommend describing what transaction receipts do, rather than listing the use cases.  (Use cases are great in design docs, but seeing the term "use case" in user docs is a peeve of mine.)  Here's a possible rewrite that also eliminates the nested bullets and fixes the punctuation and pronoun issues: 

     Transaction receipts provide clients with information that is related to the execution of a 
     transaction but should not be stored in state, such as:
       - Whether the transaction was valid
       - How the transaction changed state
       - Events of interest that occurred during execution of the transaction
       - Other transaction-family-specific execution information
     
     Transaction receipts can also provide the validator with information about transaction execution 
     without re-executing the transaction.